server/mailet/mailetcontainer-camel/src/main/java/org/apache/james/mailetcontainer/impl/JamesMailetContext.java

```diff
@@ -455,30 +456,30 @@ public void configure(HierarchicalConfiguration config) throws ConfigurationExce
             // list of supported domains that isn't localhost. If that
             // doesn't work, use the hostname, even if it is localhost.
             if (postMasterAddress.indexOf('@') < 0) {
-                String domainName = null; // the domain to use
+                Domain domainName = null; // the domain to use
                 // loop through candidate domains until we find one or exhaust
                 // the
                 // list
-                for (String dom : domains.getDomains()) {
-                    String serverName = dom.toLowerCase(Locale.US);
-                    if (!("localhost".equals(serverName))) {
-                        domainName = serverName; // ok, not localhost, so
+                Domain localhost = Domain.of("localhost");
+                for (Domain dom : domains.getDomains()) {
+                    if (!(localhost.equals(dom))) {
+                        domainName = dom; // ok, not localhost, so
                         // use it
                     }
                 }
 
                 // if we found a suitable domain, use it. Otherwise fallback to
                 // the
                 // host name.
-                postMasterAddress = postMasterAddress + "@" + (domainName != null ? domainName : domains.getDefaultDomain());
+                postMasterAddress = postMasterAddress + "@" + (domainName != null ? domainName.name() : domains.getDefaultDomain());
```

Comment:
Looks the perfect use case for Optionals, and a **toMailAddress(String)** method.

Reply:
already rewritten